File: android/app/src/main/java/com/lightningkite/kwifttemplate/shared/views/ExampleContentVG.kt
package com.lightningkite.kwifttemplate.shared.views

import android.view.View
import com.lightningkite.kwift.observables.actual.bindText
import com.lightningkite.kwift.observables.shared.StandardObservableProperty
import com.lightningkite.kwift.shared.captureWeak
import com.lightningkite.kwift.views.actual.ViewDependency
import com.lightningkite.kwift.views.actual.onClick
import com.lightningkite.kwift.views.shared.ViewGenerator
import com.lightningkite.kwifttemplate.xml.ExampleContentXml

class ExampleContentVG : ViewGenerator() {
    override val title: String get() = "Example Content"

    val number: StandardObservableProperty<Int> = StandardObservableProperty(0)

    fun increment(){
        number.value += 1
    }

    override fun generate(dependency: ViewDependency): View {
        val xml = ExampleContentXml()
        val view = xml.setup(dependency)
        xml.exampleContentIncrement.onClick(captureWeak(this) { self -> self.increment() })
        xml.exampleContentNumber.bindText(number) { it -> it.toString() }
        return view
    }
}

File: ios/Kwift Template/com/lightningkite/kwifttemplate/shared/views/ExampleContentVG.swift
//Package: com.lightningkite.kwifttemplate.shared.views
//Converted using Kwift2

import Foundation
import Kwift
import KwiftMaps



public class ExampleContentVG: ViewGenerator {
    
    
    override public var title: String {
        get {
            return "Example Content"
        }
    }
    public var number: StandardObservableProperty<Int32>
    public var chained: StandardObservableProperty<MutableObservableProperty<Int32>>
    
    public func increment() -> Void {
        number.value += 1
    }
    
    override public func generate(dependency: ViewDependency) -> View {
        var xml = ExampleContentXml()
        var view = xml.setup(dependency)
        xml.exampleContentIncrement.onClick(captureWeak(self) { (self) in 
            self.increment()
        })
        xml.exampleContentNumber.bindText(number) { (it) in 
            it.toString()
        }
        xml.chainedIncrement.onClick{ () in 
            self.chained.value.value = self.chained.value.value + 1
        }
        xml.chainedNumber.bindString(chained.flatMap{ (it) in 
            it
        }.map{ (it) in 
            it.toString()
        })
        return view
    }
    override public func generate(_ dependency: ViewDependency) -> View {
        return generate(dependency: dependency)
    }
    
    override public init() {
        let number: StandardObservableProperty<Int32> = StandardObservableProperty(0)
        self.number = number
        let chained: StandardObservableProperty<MutableObservableProperty<Int32>> = StandardObservableProperty(StandardObservableProperty(0))
        self.chained = chained
        super.init()
    }
}
 
